requirements/optional.txt

```diff
@@ -0,0 +1,16 @@
+# some of these dependencies live on a private git server. If you have the address,
+# configure git with:
+# git config --global url.https://<private hostname>/.insteadOf https://private.repo/
```

Comment:
@willbarton I think you proposed making cfgov-refresh into an installable python package, and maintaining a requirements.txt in a private repo that references a given cfgov-refresh version, along with these private repos, releasing versions of *that* repo. Let's call this new repo cfgov-everything

requirements.txt in cfgov-everything (master) might look like:

```
git+https://github.com/cfpb/cfgov-refresh.git@3.5.1
git+https://secret/CFPB/cfgov-selfregistration.git@v1.1#egg=selfregistration
...
```
Or maybe like this
```
git+https://github.com/cfpb/cfgov-refresh.git@master
git+https://secret/CFPB/cfgov-selfregistration.git@v1.1#egg=selfregistration
...
```

In the first version, On our build server, we deploy the master version of cfgov-everything, except for cfgov-refresh: we ignore the pinned version and deploy master of *that*. In the second, master of cfgov-everything implies master of cfgov-refresh, and when we create a release, we have to create releases of both repos, and update requirements.txt in the release branch of cfgov-everything to point to the new release of cfgov-refresh.

It's not terrible, especially if we automate things.

I think this is simpler, though, and I see advantages in having a single repo that describes the (code) state of the website. For one thing, it'll make it easier to communicate with groups like release management and security.

But honestly, it's just the difference between "this is cf.gov" vs. "this and this are cf.gov" both are superior to the current setup, of course ;)